### PR TITLE
[WebApp] Achievements: feature flag condition - added

### PR DESCRIPTION
**Pull request by vitto-moz**
https://www.notion.so/saladtech/WebApp-Add-Unleash-flag-91cd8a956aa54cbf9ad50dba9c048ac2?pvs=4

### CODE REVIEW

File: packages/web-app/src/components/SettingsPage.tsx [hunk 1]
```diff
@@ -28,64 +32,74 @@ export interface MenuButton {
...
-    const { onClose } = this.props
+  const handleCloseClicked = () => {
     onClose?.()
   }
 
-  private handleCloseKeyPress = (e: any) => {
-    if (e.key === 'Escape') {
-      const { onClose } = this.props
-      onClose?.()
+  const handleCloseKeyPress = useCallback(
```
Comment (rners01) [inline packages/web-app/src/components/SettingsPage.tsx:46]: why do we need `useCallback` here?

File: packages/web-app/src/components/SettingsPage.tsx [hunk 2]
```diff
@@ -28,64 +32,74 @@ export interface MenuButton {
...
     }
-  }
+  }, [handleCloseKeyPress])
+
+  const menuItems: MenuItem[] = [
+    { url: '/account/summary', text: 'Account', component: AccountContainer },
+    { url: '/account/referrals', text: 'Referrals', component: ReferralSettingsContainer },
+    { url: '/account/bonuses', text: 'Bonuses', component: BonusPageContainer },
+    isAchievementsFeatureFlagEnabled && {
+      url: '/account/achievements',
```
Comment (rners01) [inline packages/web-app/src/components/SettingsPage.tsx:67]: what would we show if user is not logged in?

File: packages/web-app/src/components/SettingsPage.tsx (hunk 1 above)
Comment (vitto-moz) [reply packages/web-app/src/components/SettingsPage.tsx:46]: this function is used as a dependency for `useEffect`
```
 useEffect(() => {
    document.addEventListener('keyup', handleCloseKeyPress)
    return () => {
      document.removeEventListener('keyup', handleCloseKeyPress)
    }
  }, [handleCloseKeyPress])

File: packages/web-app/src/components/SettingsPage.tsx (hunk 2 above)
Comment (vitto-moz) [reply packages/web-app/src/components/SettingsPage.tsx:67]: image

**Review by Maks19 — APPROVED** [review]

**Pull request merged**